Sources/TuistKit/Graph/GraphLoader.swift

```diff
@@ -42,13 +42,26 @@ class GraphLoader: GraphLoading {
         return graph
     }
 
-    func loadWorkspace(path: AbsolutePath) throws -> Graph {
+    func loadWorkspace(path: AbsolutePath) throws -> (Workspace, Graph) {
         let cache = GraphLoaderCache()
         let circularDetector = GraphCircularDetector()
         let workspace = try modelLoader.loadWorkspace(at: path)
-        let projects = try workspace.projects.map { (projectPath) -> (AbsolutePath, Project) in
-            try (projectPath, Project.at(projectPath, cache: cache, circularDetector: circularDetector, modelLoader: modelLoader))
+
+        func traverseProjects(element: Workspace.Element) throws -> [(AbsolutePath, Project)] {
+            switch element {
+            case .file(path: _):
+                break
+            case .group(name: _, contents: let contents):
+                return try contents.flatMap(traverseProjects)
+            case let .project(path: path):
+                return [try (path, Project.at(path, cache: cache, circularDetector: circularDetector, modelLoader: modelLoader))]
+            }
+
+            return []
         }
+
+        let projects = try workspace.contents.flatMap(traverseProjects)
+
         let entryNodes = try projects.flatMap { (project) -> [TargetNode] in
```

Comment:
Reading this now, I think using `$0`, `.1` , `.0` makes the code hard to read. Would you mind updating it to use real names?